src/main/scala/com/whisk/k8s/resolver/KubernetesClient.scala

```diff
@@ -92,6 +97,29 @@ class KubernetesClient(client: Service[Request, Response]) extends Closable {
     } yield {}
   }
 
+  /**
+    * Get content from the Kubernetes. If it's not chunked - get the content directly,
+    * otherwise collect the content and then give it back to decoder as
+    * circle.io can't process stream content.
+    */
+  private def getContent(response: Response): Future[Buf] = {
+    if (response.isChunked) {
+
+      //TODO: Trampoline this if recursion goes too deep
+      def read(reader: Reader[Chunk], lastContent: Buf): Future[Buf] = reader.read().flatMap {
```

Comment:
You can use `Reader.readAll(response.reader)` here =) And maybe, we can use this construction for each type of response.

Reply:
`Reader.readAll(response.reader)` could be used in case we have chunked response, otherwise content will be populated with original response data.